enterprise/backend/src/metabase_enterprise/data_editing/api.clj

```diff
@@ -313,6 +314,35 @@
   (let [include-cols [:token :table_id :creator_id]]
     {:tokens (t2/select (into [:table_webhook_token] include-cols) :table_id table-id)}))
 
+(api.macros/defendpoint :get "/action"
+  "Lists all available actions and their parameters for a context.
+  Only supported context is from a dashboard editable right now, the dashcard-id must be supplied
+  as a query parameter.
+  Only returns actions the user has access to.
+  Return type is the same as GET /api/action"
+  [{:as _route-params}
+   {:keys [dashcard-id]} :- [:map
+                             [:dashcard-id ms/PositiveInt]]]
+  (let [dashcard        (api/check-404 (t2/select-one :model/DashboardCard :id dashcard-id))
+        table-id        (:table_id (:visualization_settings dashcard))
```

Comment:
Rather than going via the table, I think we should go via the `editable` given by `card_id`.

```clojure
(let [editable (t2/select-one :model/Card :type "model" :id (:card_id dashcard))] 
  ;; we could check that `:display` is `table-editable` here, but from discussion with Sanya 
  ;; there  could be use cases for row actions on read-only tables as well.
  ...)
```

This is more straightforward, and is also future-proof for editable join tables.